release/models/system/openconfig-system-controlplane.yang

```diff
@@ -0,0 +1,181 @@
+  yang-version "1";
+
+  namespace "http://openconfig.net/yang/system-controlplane";
+  prefix "oc-sys-copp";
+
+  import openconfig-extensions { prefix oc-ext; }
+  import openconfig-system { prefix oc-sys; }
+  import openconfig-acl { prefix oc-acl; }
+  import openconfig-qos { prefix oc-qos; }
+
+  organization
+    "OpenConfig working group";
+  contact
+    "www.openconfig.net";
+
+  description
+    "This module adds configuration and operational state relating to
+    policies for traffic destined to the system's control-plane.
+    Particularly, it allows for mechanisms to:
+     - apply an ACL that forwards or drops traffic towards the control-plane.
+     - classify traffic that is destined to the control-plane according to
+       a QoS classifier.
+     - schedule traffic that has been forwarded towards the control-plane,
+       to allow for policies such as rate limits to be applied.
+    The configured policies apply generically to all control-planes that
+    exist within the system, and should be mapped to the internal interfaces
+    via which packets are forwarded to control-plane modules.
+    When a packet is received at an input interface - it is classified into a
+    forwarding group which drains to a specific queue. If this input mapping
+    is sufficient, the CPU-facing interface uses the specified scheduler
+    to determine how to drain queues. If more granular remapping is required
+    (e.g., to classify control-plane traffic more granularly), a user specifies
+    an alternate classifier that is used to reclassify traffic into
+    a new set of forwarding-groups (and hence queues) that can subsequently
+    be scheduled by the specified scheduler.
+    The specified control-plane ACL is applied to traffic received by the
+    control-plane of the system.";
+
+
+  oc-ext:openconfig-version "0.1.0";
+  oc-ext:catalog-organization "openconfig";
+  oc-ext:origin "openconfig";
+
+  revision "2021-08-02" {
+    description
+      "Initial revision.";
+    reference "0.1.0";
+  }
+
+  grouping system-controlplane-top {
+    description
+      "Top-level structural grouping for control-plane traffic policies.";
+
+    container control-plane-traffic {
+      description
+        "Policies and configuration relating to the traffic destined towards
+        the system control-plane.";
+
+      container config {
+        description
+          "Configuration parameters relating to control-plane traffic.";
+
+        uses system-controlplane-config;
+      }
+
+      container state {
+        config false;
+        description
+          "Operational state parameters relating to control-plane traffic.";
+
+        uses system-controlplane-config;
+      }
+
+      container acls {
+        description
+          "Access control lists (ACLs) that are to be applied on traffic
+          destined for the control-plane. An ACL per traffic type
+          can be applied.";
+
+        list acl {
+          key "type";
```

Comment:
Do we want to consider a per network-instance control-plane ACL configuration?

See https://www.arista.com/en/um-eos/eos-acls-and-route-maps#xx1302661

Reply:
@dplore The more I think about it the more we likely need to make this a per network-instance collection of `acl-set` leafrefs. Within the `ipv4` and `ipv6` containers, there is no indication of which network-instance to refer to. It also doesn't make sense for `acl-sets` to be NI aware as it can be associated per interface which is NI inherited. Therefore this control-plane configuration would have to be NI aware.

One side note is if the intention is to apply the ACL across all VRFs (with an `acl-set` that doesn't have any IP addresses, usually default ACLs) then that configuration should be an replacement of whatever default ACL that comes on a switch and there should be a separate per VRF collection which would override this default.

Reply:
I'm not sure that this is something that is generally compatible.

* Cisco IOS XR LPTS does not match on a [VRF](https://www.cisco.com/c/en/us/td/docs/iosxr/cisco8000/ip-addresses/75x/b-ip-addresses-cg-8k-75x/implementing_lpts.html#Cisco_Concept.dita_906ea9e6-b511-4c74-a3b4-50819baf860d).
* JUNOS allows `fxp0` to be assigned to one routing-instance, and firewall filters don't seem to support VRF matches [doc](https://www.juniper.net/documentation/us/en/software/junos/routing-policy/topics/concept/firewall-filter-match-conditions-for-ipv4-traffic.html)
* Nokia SROS - see [here](https://infocenter.nokia.com/public/7750SR150R4A/index.jsp?topic=/com.sr.system.mgmt/html/getstarted.html) does not have a binding of `cpm-filter` to a particular VRF.

To this end, I don't propose to extend the model to cover VRFs. If this is something that we want to add in the future, we could add a VRF list that gives us ACL-per-VRF and leave a single pair (ingress/egress) of references here for "all VRFs". The new list would be overrides and/or mutually exclusive with this leaf (if we decide to add it based on device compatibility).

Reply:
Nokia SR OS has a VRF as a match criteria inside the cpm-filter:
https://documentation.nokia.com/sr/22-10-2/books/system-management/security-system-management-1.html#ai9exj5yak__ai9exj5yaq